grafana/resource_machine_learning_outlier_detector_test.go

```diff
@@ -183,14 +149,6 @@ func TestAccResourceInvalidMachineLearningOutlierDetector(t *testing.T) {
 				Config:      machineLearningOutlierDetectorInvalid,
 				ExpectError: regexp.MustCompile(".*datasource_type.*"),
 			},
-			{
-				Config:      machineLearningOutlierDetectorMissingDatasourceIDOrUID,
-				ExpectError: regexp.MustCompile(".*datasource_id or datasource_uid.*"),
```

Comment:
You should keep the test. Terraform should throw an error and it's nice to validate that it still works. Same for `algorithm`

Reply:
Ok. I thought it might be redundant checking TF's behaviour.